Ical.Net/DataTypes/CalDateTime.cs

```diff
@@ -1,130 +1,188 @@
+    /// <summary>
+    /// Gets the current date/time in the Coordinated Universal Time (UTC) timezone.
+    /// </summary>
+    public static CalDateTime UtcNow => new CalDateTime(DateTime.UtcNow, UtcTzId, true);
 
+    /// <summary>
+    /// This constructor is required for the SerializerFactory to work.
+    /// </summary>
     public CalDateTime() { }
 
+    /// <summary>
+    /// Creates a new instance of the <see cref="CalDateTime"/> class.
+    /// </summary>
+    /// <param name="value"></param>
     public CalDateTime(IDateTime value)
     {
-        Initialize(value.Value, value.TzId, null);
+        if (value.HasTime)
+            Initialize(DateOnly.FromDateTime(value.Value), TimeOnly.FromDateTime(value.Value), value.TzId);
+        else
+            Initialize(DateOnly.FromDateTime(value.Value), null, value.TzId);
     }
 
-    public CalDateTime(DateTime value) : this(value, null) { }
+    /// <summary>
+    /// Creates a new instance of the <see cref="CalDateTime"/> class
+    /// and sets the <see cref="TzId"/> to "UTC" if the <paramref name="value"/>
+    /// has <see cref="DateTimeKind.Utc"/>, otherwise it will be left as <see langword="null"/>.
+    /// <para/>
+    /// The timezone will be set to UTC if the <paramref name="value"/> has <see cref="DateTimeKind.Utc"/>.
+    /// Else, the timezone will be <see langword="null"/>.
+    /// </summary>
+    /// <param name="value"></param>
+    /// <param name="hasTime">Set to <see langword="true"/> (default), if the <see cref="DateTime.TimeOfDay"/> must be included.</param>
+    public CalDateTime(DateTime value, bool hasTime = true) : this(value, value.Kind == DateTimeKind.Utc ? UtcTzId : null, hasTime)
+    { }
 
     /// <summary>
-    /// Specifying a `tzId` value will override `value`'s `DateTimeKind` property. If the time zone specified is UTC, the underlying `DateTimeKind` will be
-    /// `Utc`. If a non-UTC time zone is specified, the underlying `DateTimeKind` property will be `Local`. If no time zone is specified, the `DateTimeKind`
-    /// property will be left untouched.
+    /// Creates a new instance of the <see cref="CalDateTime"/> class using the specified timezone.
     /// </summary>
-    public CalDateTime(DateTime value, string tzId)
+    /// <param name="value"></param>
+    /// <param name="tzId">The specified value will override value's <see cref="DateTime.Kind"/> property.
+    /// If the timezone specified is UTC, the underlying <see cref="DateTime.Kind"/> will be
+    /// <see cref="DateTimeKind.Utc"/>. If a non-UTC timezone or no timezone is specified, the
+    /// <see cref="DateTimeKind.Unspecified"/> will be used. A timezone of <see langword="null"/> represents
+    /// the system's local timezone.
+    /// </param>
+    /// <param name="hasTime">Set to <see langword="true"/> (default), if the <see cref="DateTime.TimeOfDay"/> must be included.</param>
+    public CalDateTime(DateTime value, string? tzId, bool hasTime = true)
```

Comment:
Having both, the `Kind` in the `DateTime` value, as well as the `tzId`, causes redundancy, that's not easy to understand. What, if we remove this constructor altogether? The user could easily call `new CalDateTime(value).ToTimeZone(tzId)`, with way less ambiguity.

Reply:
This will be back on the agenda when we implement floating date/time. See #647